### ISSUE TITLE
Cookiebot - prevent cookies from firing before consent.

### ISSUE THREAD

**Issue by aileenrose** [sourcegraph/about#2274]
TLDR:  Cookiebot's latest cookie scan reported that some cookies may be firing before consent.   Currently, most third-party cookies ( i.e. CrazyEgg, LinkedIn) are managed by Google Tag Manager which includes Cookiebot's own consent management cookies.   There are other cookies (i.e. Hubspot) that are being set directly via gatsby.js.  To ensure that we are respecting the user's consent, we can use:
- Google Consent Mode to control cookies managed in Google Tag Manager.  
- Cookiebot to automatically control all other cookies written by third-party apps and gatsby plug-ins in gatsby.js.

Background details:
To improve privacy and be more in line with GDPR and the ePrivacy Directive, we need to make an update to how we manage cookie consent on the website. Google recently released Consent Mode which will honor a user's consent*  for its products including Google Analytics, Google Ads, and Google Tag Manager.   
*Note: Google Consent Mode does not manage or store consent, it integrates with Cookiebot which does.

What is the impact on our advertising?  Most importantly, we are respecting a user's privacy based on their consent.  In Consent Mode, Google will not store any visitor data that is used for remarketing or personalized advertising. In effect, the target audience will be smaller but we only advertise to users that have consented.  If the user has not consented, Google Analytics/Ads will still record aggregate level data for tracking conversions.

What about other advertising?  As we are expanding our ad programs into LinkedIn, and possibly Twitter, Reddit, etc., we are managing all their tracking configurations in Google Tag Manager.  So we will be covered here since Consent Mode works with Google Tag Manager+Cookiebot and these advertising platforms will not store visitor data unless a user has consented.

What about the [UTM parameters tracking project](https://docs.google.com/document/d/1z1rpQuLpq1BCV-21JZSd4ORCfVWYTxH9iGvso00WEZo/edit?ts=5fe0e2a2#)?  Since we will be using cookies to store the information of a user’s first visit to the website, we will also need to respect the user’s consent.  If a user has not provided consent, we cannot track their initial URL ('First source URL').  Note, we will still be able to capture UTM data for any visitors that clicked on ads and converted in the same session.  This information is automatically recorded in the Hubspot field:  'First page seen'.  If a user clicks on an ad, does not consent, leaves the website, returns organically or through another ad, and finally converts on a Hubspot form, 'First page seen' will capture the session's first URL.  The 'First source URL' will be null.

Other cookies?  A couple of days ago, Cookiebot notified me on their last scan that we have some cookies are being written before consent is submitted.  Third-party apps like Hubspot and added functionality from Gatsby.js plug-ins are managed on the website, not by Google Tag Manager (they are configured directly in Gatsby). To cover cookies written by other apps and website functions i.e. not maintained in Google Tag Manager, we can respect consent using Cookiebot’s automatic cookie blocking mode - a configuration that rolled out last year which I recently learned about.

**Comment by limitedmage** [comment]
One simple thing we can start with is switching all our embedded YouTube videos to YouTube's no cookie domain (from `www.youtube.com` to `www.youtube-nocookie.com`). That way YouTube won't drop any cookies on the user when they visit our site.

**Comment by aileenrose** [comment]
> One simple thing we can start with is switching all our embedded YouTube videos to YouTube's no cookie domain (from `www.youtube.com` to `www.youtube-nocookie.com`). That way YouTube won't drop any cookies on the user when they visit our site.
@limitedmage  excellent - I'll do a quick Sourcegraph search!  Thanks!

**Comment by aileenrose** [comment]
Switch to no cookies for embedded YouTube links:
https://github.com/sourcegraph/about/pull/2275

**Comment by aileenrose** [comment]
To implement:
1) Configure Cookiebot's automatic cookie blocking.
https://support.cookiebot.com/hc/en-us/articles/360009192739-Google-Tag-Manager-and-Automatic-cookie-blocking
 -  `gatsby-plugin-google-tagmanager` will need to be removed since the plugin does not allow mark up of the <script> tag.
 -  Add the GTM script to the Header.js which will include the <script> tag `data-cookieconsent="ignore"`
 -  Add Cookiebot script below GTM script
 -  In GTM, update the ***Cookiebot CMP**  tag from all pages.

2) Set up Google Consent Mode which respects user's consent.   Cookiebot recently announced an integration with Google Consent Mode. **This mode is still in Beta** and there may be changes when it goes into production.
